src/content_cao.cpp

```diff
@@ -1166,8 +1166,19 @@ void GenericCAO::step(float dtime, ClientEnvironment *env)
 	} else {
 		v3f lastpos = pos_translator.vect_show;
 
-		if(m_prop.physical)
-		{
+		if(m_prop.physical) {
+			if (m_knockback_timer > 0.0f) {
+				m_knockback_timer -= dtime;
+				// If the player is dead, stop the movement.
+				if (m_hp <= 0) {
+					m_velocity = v3f(0, 0, 0);
+				}
+					// If the timer is finished go down the player.
+				else if (m_knockback_timer <= 0.0f) {
+					m_velocity = v3f(0, -9.81 * BS, 0);
```

Comment:
`v3f(0, -9.81 * BS, 0);` should be the acceleration applied throughout the flight of theplayer, this is acceleration not velocity.